### PR TITLE
feat: integrate new stream types

### PR DESCRIPTION
**Pull request by bbaldino**
This PR gets rid of the old track classes and changes things to use the new stream types.

### CODE REVIEW

File: src/device/device-management.ts [hunk 1]
```diff
@@ -70,20 +72,18 @@ export async function createCameraTrack<T extends LocalTrack>(
...
 
 /**
- * Creates a microphone audio track.
+ * Creates a LocalMicrophoneStream with the given constraints.
  *
- * @param constructor - Constructor for the local microphone track.
  * @param constraints - Audio device constraints.
  * @returns A LocalTrack object or an error.
  */
-export async function createMicrophoneTrack<T extends LocalTrack>(
```
Comment (mccarthytyler) [inline src/device/device-management.ts:83]: The `<T extends LocalTrack>` may need to stay in, I think it's used in some cases to create a custom type of stream wrapper.  Maybe `<T extends LocalMicrophoneStream>`?

**Review by mccarthytyler — APPROVED** [review]
Looks good, just one question.

File: src/device/device-management.ts (hunk 1 above)
Comment (bbaldino) [reply src/device/device-management.ts:83]: The plan is to do this differently, where extend of extending these tracks (which then means it's disconnected from the rest of the hierarchy)--they'd instead create and wrap this track, so if all goes well then the `T extends XXX` thing will not be needed.

Comment (mccarthytyler) [reply src/device/device-management.ts:83]: Cool, sounds good.

**Pull request merged**